src/main/java/com/etsy/statsd/profiler/server/ProfilerServer.java

```diff
@@ -21,9 +27,17 @@
      * @param activeProfilers The active profilers
      * @param port The port on which to bind the server
      */
-    public static void startServer(final Map<String, ScheduledFuture<?>> activeProfilers, final int port) {
-        HttpServer server = vertx.createHttpServer();
-        server.requestHandler(RequestHandler.getMatcher(activeProfilers));
-        server.listen(port);
+    public static void startServer(final Map<String, ScheduledFuture<?>> runningProfilers, final Map<String, Profiler> activeProfilers, final int port, final AtomicReference<Boolean> isRunning, final LinkedList<String> errors) {
+        final HttpServer server = vertx.createHttpServer();
+        server.requestHandler(RequestHandler.getMatcher(runningProfilers, activeProfilers, isRunning, errors));
+        server.listen(port, new Handler<AsyncResult<HttpServer>>() {
+            @Override
+            public void handle(AsyncResult<HttpServer> event) {
+                if (event.failed()) {
+                    server.close();
+                    startServer(runningProfilers, activeProfilers, port + 1, isRunning, errors);
```

Comment:
We could use the built-in `java.util.logging.Logger`.  It looks like slf4j and log4j are getting pulled in transitively, but I had avoided them for precisely the reason you mention.